File: cpp/StorageKomponen.cpp
// StorageKomponen.cpp
#pragma once
#include <iostream>
#include <string>
#include "Ram.cpp"
#include "Harddrive.cpp"

using namespace std;

class StorageKomponen : public virtual Ram, public virtual Harddrive
{
private:
    int kecepatanBaca;
    int kecepatanTulis;

public:
    StorageKomponen() {}

    StorageKomponen(int kecepatanBaca, int kecepatanTulis, int kapasitasGBRam, string ddr, string merkRam, string namaRam,
                    int kapasitasGBHDD, string tipeDrive, string merkHDD, string namaHDD, string model, int konsumsiDayaWatt)
        : Komponen(merkRam, namaRam), Ram(kapasitasGBRam, ddr, merkRam, namaRam),
          Harddrive(kapasitasGBHDD, tipeDrive, merkHDD, namaHDD, model, konsumsiDayaWatt),
          kecepatanBaca(kecepatanBaca), kecepatanTulis(kecepatanTulis) {}

    void setKecepatanBaca(int kecepatanBaca) { this->kecepatanBaca = kecepatanBaca; }
    void setKecepatanTulis(int kecepatanTulis) { this->kecepatanTulis = kecepatanTulis; }

    int getKecepatanBaca() const { return kecepatanBaca; }
    int getKecepatanTulis() const { return kecepatanTulis; }

    ~StorageKomponen() {}
};

File: py/StorageKomponen.py
from Ram import Ram
from Harddrive import Harddrive

# StorageKomponen.py
class StorageKomponen(Ram, Harddrive):
    def __init__(self, kecepatanBaca=0, kecepatanTulis=0, 
                 kapasitasGBRam=0, ddr="", merkRam="", namaRam="",
                 kapasitasGBHDD=0, tipeDrive="", merkHDD="", namaHDD="", 
                 model="", konsumsiDayaWatt=0):
        Ram.__init__(self, kapasitasGBRam, ddr, merkRam, namaRam)
        Harddrive.__init__(self, kapasitasGBHDD, tipeDrive, merkHDD, namaHDD, model, konsumsiDayaWatt)
        self.kecepatanBaca = kecepatanBaca
        self.kecepatanTulis = kecepatanTulis
    
    def setKecepatanBaca(self, kecepatanBaca):
        self.kecepatanBaca = kecepatanBaca
    
    def setKecepatanTulis(self, kecepatanTulis):
        self.kecepatanTulis = kecepatanTulis
    
    def getKecepatanBaca(self):
        return self.kecepatanBaca
    
    def getKecepatanTulis(self):
        return self.kecepatanTulis
